### DISCUSSION TITLE
within design without repetition of variables

### DISCUSSION THREAD

**Discussion by HeRm4nV** [jspsych/jsPsych#1495] [Q&A]
Hi,

I'm doing a little design with 2 within variables, but i want only 2 trials for person (1 question -> 2 trials), first trial using one variable of first within variable and other from the second, and the second trial using the remainings variables. I currently have this code working:

```
var within_var = {
  condition_01: ["breast-cancer", "down-syndrome"],
  condition_02: ["20", "40"]
}

//creating design variables for timeline
var full_design = [];
var first = true;
Object.keys(within_var).forEach(function(key) {
  var count = 0;
  jsPsych.randomization.repeat(within_var[key],1).forEach( function(value, index, array) {
    if (first) {
      full_design.push({});
    }
    full_design[count][key] = value;
    count += 1;
  });
  first = false;
});
```

But I would like to know if there is any other simpler way like the one shown to create a full_design or if it isn't currently implemented:

`var full_design = jsPsych.randomization.factorial(within_var, 1);`

**Comment by jodeleeuw** [comment]
Hi @HeRm4nV,

Currently the `jsPsych.ranomization.factorial()` method only produces full factorial designs. It would be awesome to have a more diverse set of randomization functions that could create designs like yours. It's not a high priority on our development list right now because it can just be done in non-jspsych JS code, but if you're inspired to create such methods and submit a pull request I'd welcome the contribution with enthusiasm!
